FOUNDATIONS.md

```diff
@@ -6,10 +6,14 @@ ensure the success of future improvements (especially type systems).
 **Definitions**
 
 * Block: A block is a chunk of an IPLD DAG, encoded in a format. Blocks have CIDs.
-* Fragment: A piece of an IPLD DAG. Blocks contain fragments.
 * Node: A node is a *point* in an IPLD DAG (object, array, number, etc.).
-* Link: A link is an IPLD Node that points to another IPLD Node.
-* Path: A paths a human readable pointer to an IPLD Node.
+  Many nodes can exist encoded inside one Block.
+* Link: A link is a kind of IPLD Node that points to another IPLD Node.
+* Path Segment: A path segment a piece of information that describes a move from
+  one Node to another Node (effectively, either a map key or a list index).
```

Comment:
```suggestion
  the current Node to another Node (effectively, either a map key or a list index).
```
A **single** Path Segment defines a move **relative** to the current position.

Reply:
I'm not sure "the current" is necessary, if a Path is made up of many segments, they each relate to different Nodes, not "the current". But it does need to be tightened so it doesn't sound like arbitrary hops constitute a "Path Segment". Maybe "one Node to an adjacent Node", or "one Node to a directly connected Node"?

Also .. are we leaving the advanced layout case out of it for now, pretending that Paths only operate on the data model and that's all?

Reply:
Re last point about advanced layouts addressed in #157 but probably suggests that #157 _shouldn't_ be entirely in a separate doc, will comment there.

Reply:
> if a Path is made up of many segments, they each relate to different Nodes

Yes, but each Path Segment itself points from its current node to another. So I'm in favour of "the current Node".

Reply:
So in the context of a `Path Segment` (highlighting the singular), you have `./a`, which describes the move from `.` (the current node) to `a`, the root being where we are currently with a relative movement to `a`.

Reply:
I also thing "current" is a little weird without having been introduced, but I'll massage this a bit more to try to incorporate the idea.

Reply:
Another thing that would be nice to capture here is that the step is, er, "downward".  E.g., `".."` is not a path segment that means "go up one", even though people might expect that because of the history of unixy filesystemy behavior.  But this is also tricking without having introduced "up" vs "down" directions.

It seems like there's a limit on how much meaning we can pack into these short introducers here.  We should probably start thinking about what the tersest expressions of these concepts are, and then linking to fuller specs elsewhere (maybe even later in the same document, if we're trying to make this one fairly freestanding).